File: DuDoong-Api/src/main/java/band/gosrock/api/event/service/CreateEventUseCase.java
package band.gosrock.api.event.service;


import band.gosrock.api.common.UserUtils;
import band.gosrock.api.event.model.dto.request.CreateEventRequest;
import band.gosrock.api.event.model.dto.response.EventResponse;
import band.gosrock.api.event.model.mapper.EventMapper;
import band.gosrock.common.annotation.UseCase;
import band.gosrock.domain.domains.event.domain.Event;
import band.gosrock.domain.domains.event.service.EventService;
import band.gosrock.domain.domains.host.service.HostService;
import lombok.RequiredArgsConstructor;
import org.springframework.transaction.annotation.Transactional;

@UseCase
@RequiredArgsConstructor
public class CreateEventUseCase {
    private final UserUtils userUtils;
    private final HostService hostService;
    private final EventService eventService;
    private final EventMapper eventMapper;

    @Transactional
    public EventResponse execute(CreateEventRequest createEventRequest) {
        final Long userId = userUtils.getCurrentUserId();
        // 슈퍼 호스트 이상만 공연 생성 가능
        hostService.validateManagerHostUser(createEventRequest.getHostId(), userId);
        final Event event = eventMapper.toEntity(createEventRequest);
        return EventResponse.of(eventService.createEvent(event));
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/event/service/CreateEventUseCase.kt
package band.gosrock.api.event.service

import band.gosrock.api.event.model.dto.request.CreateEventRequest
import band.gosrock.api.event.model.dto.response.EventResponse
import band.gosrock.api.event.model.mapper.EventMapper
import band.gosrock.common.annotation.UseCase
import band.gosrock.domain.domains.event.service.EventService
import band.gosrock.domain.domains.host.service.HostService
import org.slf4j.LoggerFactory
import org.springframework.transaction.annotation.Transactional

@UseCase
class CreateEventUseCase(
    private val hostService: HostService,
    private val eventService: EventService,
    private val eventMapper: EventMapper
) {
    private val log = LoggerFactory.getLogger(CreateEventUseCase::class.java)

    @Transactional
    fun execute(userId: Long, createEventRequest: CreateEventRequest): EventResponse {
        log.info("[CreateEventUseCase][execute] 이벤트 생성 userId={} hostId={}", userId, createEventRequest.hostId)
        // 슈퍼 호스트 이상만 공연 생성 가능
        hostService.validateManagerHostUser(createEventRequest.hostId!!, userId)
        val event = eventMapper.toEntity(createEventRequest)
        return EventResponse.of(eventService.createEvent(event))
    }
}
